### ISSUE TITLE
Registering services crashes with None args

### ISSUE THREAD

**Issue by bit-pirate** [robotics-in-concert/rocon_multimaster#173]
```
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/rocon_gateway/gateway.py", line 22, in <module>
    gateway.spin()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rocon_gateway/gateway.py", line 74, in spin
    self.watcher_thread.start()
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rocon_gateway/watcher_thread.py", line 74, in start
    self._gateway.update_pulled_interface(connections, remote_gateway_hub_index)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rocon_gateway/gateway.py", line 196, in update_pulled_interface
    new_registration = self.master.register(registration)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rocon_gateway/master_api.py", line 336, in register
    node_master.registerService(registration.connection.rule.name, registration.connection.type_info, registration.connection.xmlrpc_uri)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rosgraph/masterapi.py", line 315, in registerService
    return self._succeed(self.handle.registerService(self.caller_id, service, service_api, caller_api))
  File "/usr/lib/python2.7/xmlrpclib.py", line 1224, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1572, in __request
    allow_none=self.__allow_none)
  File "/usr/lib/python2.7/xmlrpclib.py", line 1085, in dumps
    data = m.dumps(params)
  File "/usr/lib/python2.7/xmlrpclib.py", line 632, in dumps
    dump(v, write)
  File "/usr/lib/python2.7/xmlrpclib.py", line 654, in __dump
    f(self, value, write)
  File "/usr/lib/python2.7/xmlrpclib.py", line 658, in dump_nil
    raise TypeError, "cannot marshal None unless allow_none is enabled"
TypeError: cannot marshal None unless allow_none is enabled
```

Restarting the whole thing makes the error disappear.

**Comment by stonier** [comment]
That's an odd one - how to reproduce this? Is this on the pc, netbook, starting the simple tutorials or a full turtlebot?

**Comment by bit-pirate** [comment]
I experienced this on both my workstation and on the turtlebot netbook. I'm not sure yet, how to reproduce this.

**Comment by stonier** [comment]
What were you launching exactly?

**Comment by bit-pirate** [comment]
`rocon_launch turtlebot_bringup bringup.concert`

**Comment by stonier** [comment]
Ok - probably something there is going wrong with timing as turtlebot brings up a significant amount of things at once.

**Comment by stonier** [comment]
Similar errors have appeared for people when params were set as empty strings...xmlrpc barfs. That particular bug has been fixed, but this is coming not from params.

**Comment by stonier** [comment]
Can't reproduce from my pc, but added some [error checking](https://github.com/robotics-in-concert/rocon_multimaster/commit/6cd6ff6df9a942a2c56bb9144668083ec9c08317) so we can see what is getting passed through as None when we catch it next.

**Comment by stonier** [comment]
Happened for both invite and platform_info with the type_info part. This is on the public master's gateway side when it pulls those services.

```
[ERROR] [WallTime: 1376269612.754051] Gateway : tried to register a service with type_info set to None [/app_manager/platform_info, None, http://192.168.10.123:40982/]
```

**Comment by stonier** [comment]
More interesting data - you can view some old standalone services floating around with null components:

```
redis 127.0.0.1:6380> smembers rocon:TurtleBot85b18761cb354f8fb77073a765ca2fcb:advertisements
1) "[\"service\", \"/TurtleBot85b18761cb354f8fb77073a765ca2fcb/list_apps\", \"/app_manager\", \"rosrpc://192.168.10.123:48253\", \"http://192.168.10.123:40982/\"]"
2) "[\"service\", \"/TurtleBot85b18761cb354f8fb77073a765ca2fcb/invite\", \"/app_manager\", \"rosrpc://192.168.10.123:48253\", \"http://192.168.10.123:40982/\"]"
3) "[\"service\", \"/app_manager/platform_info\", \"/app_manager\", null, \"http://192.168.10.123:40982/\"]"
4) "[\"service\", \"/TurtleBot85b18761cb354f8fb77073a765ca2fcb/platform_info\", \"/app_manager\", \"rosrpc://192.168.10.123:48253\", \"http://192.168.10.123:40982/\"]"
5) "[\"publisher\", \"/app_manager/app_list\", \"/app_manager\", \"rocon_app_manager_msgs/AppList\", \"http://192.168.10.123:40982/\"]"
6) "[\"publisher\", \"/TurtleBot85b18761cb354f8fb77073a765ca2fcb/app_list\", \"/app_manager\", \"rocon_app_manager_msgs/AppList\", \"http://192.168.10.123:40982/\"]"
7) "[\"service\", \"/TurtleBot85b18761cb354f8fb77073a765ca2fcb/status\", \"/app_manager\", \"rosrpc://192.168.10.123:48253\", \"http://192.168.10.123:40982/\"]"
```

**Comment by stonier** [comment]
Protecting the services from initialising in parallel (standalone initialisation interrupted by gateway initialisation).

https://github.com/robotics-in-concert/rocon_app_platform/commit/98bdef422f95c24fd45cd7f585977ec38381b22b

**Comment by stonier** [comment]
Its back. If you see this - please check the redis service lists as illustrated in the post two above.

**Comment by stonier** [comment]
Absolutely can't reproduce it, locally or on the turtlebot laptop.

But, I think I found the logic bug which causes it to happen. There's a small amount of time inbetween master.get_connection_state and rostopic.get_topic_type. If the topic should disappear, the latter will produce 'None'. 

I added a fix in branch `marshalling_none`. But lets see if we can reproduce it and then apply this fix to see if it works.
